torchtitan/utils.py

```diff
@@ -117,15 +117,15 @@ def init_distributed(job_config):
         os.makedirs(dump_dir, exist_ok=True)
         _warn_overwrite_env(TRACE_FILE, f"{dump_dir}/rank_")
 
-    torch.distributed.init_process_group(
-        "nccl", timeout=timedelta(seconds=job_config.comm.init_timeout_seconds)
-    )
-
     # to mitigate the memory issue that collectives using
     # async_op=True hold memory longer than they should
     # such as those in tensor parallelism
     os.environ["TORCH_NCCL_AVOID_RECORD_STREAMS"] = "1"
```

Comment:
This env var is read at PGNCCL construction time. If we set it after `init_process_group`, then technically the default PG does not see this env var as set. This is fine for us before because TP PGs are initialized after this point.

Reply:
This is a nice catch though.